arangod/RestServer/LanguageCheckFeature.cpp

```diff
@@ -56,70 +56,113 @@ arangodb::Result readLanguage(arangodb::ArangodServer& server,
       return TRI_ERROR_INTERNAL;
     }
     VPackSlice defaultSlice = content.get("default");
-    if (!defaultSlice.isString()) {
+    VPackSlice icuSlice = content.get("icu-language");
+
+    // both languages are specified in files
+    if (defaultSlice.isString() && icuSlice.isString()) {
+      return TRI_ERROR_INTERNAL;
+    }
+
+    if (defaultSlice.isString()) {
+      defaultLanguage = defaultSlice.stringView();
+    } else if (icuSlice.isString()) {
+      icuLanguage = icuSlice.stringView();
+    } else {
       return TRI_ERROR_INTERNAL;
     }
-    found = defaultSlice.copyString();
+
   } catch (...) {
     // Nothing to free
     return TRI_ERROR_INTERNAL;
   }
 
-  language = found;
-  LOG_TOPIC("c499e", TRACE, arangodb::Logger::CONFIG)
-      << "using default language: " << found;
+  if (!defaultLanguage.empty()) {
+    LOG_TOPIC("c499e", TRACE, arangodb::Logger::CONFIG)
+        << "using default language: " << defaultLanguage;
+  } else {
+    LOG_TOPIC("c490e", TRACE, arangodb::Logger::CONFIG)
+        << "using icu language: " << icuLanguage;
+  }
 
   return TRI_ERROR_NO_ERROR;
 }
 
 /// @brief writes the default language to file
 ErrorCode writeLanguage(arangodb::ArangodServer& server,
-                        std::string const& language) {
+                        std::string const& lang, bool isDefaultLang) {
   auto& databasePath = server.getFeature<arangodb::DatabasePathFeature>();
   std::string filename = databasePath.subdirectoryName("LANGUAGE");
 
   // generate json
   VPackBuilder builder;
   try {
     builder.openObject();
-    builder.add("default", VPackValue(language));
+    if (isDefaultLang) {
+      builder.add("default", VPackValue(lang));
```

Comment:
Maybe move "default" and "icu-language" to StaticStrings or at least to local constant? As these are used multiple times.

Reply:
fixed